pkg/operator/sync.go

```diff
@@ -14,6 +36,16 @@ func syncControllers(c serviceCAOperator, operatorConfig *operatorv1.ServiceCA)
 		return err
 	}
 
+	// Remove resources related to the 4.4 controller deployment at most once. These
+	// resources don't constitute an operational concern, so it is not necessarily to
+	// monitor for their presence.
+	//
+	// The 4.4 deployment does have an operational impact, and is continually monitored
+	// for removal via a goroutine started in RunOperator.
+	if err := once.Do(func() error { return cleanupUnifiedDeployment(c) }); err != nil {
```

Comment:
why is there no loop around these like it is in the other case?

Reply:
The enclosing is the operator sync loop.

Reply:
The combination with `sync.Once` confused me. Anyway, seems to be good enough.